src/client/datascience/jupyter/kernels/cellExecution.ts

```diff
@@ -131,26 +136,34 @@ export class CellExecution {
         }
 
         this.initPromise = this.enqueue();
+        workspace.onDidCloseTextDocument(
+            (e) => {
+                // If the cell is deleted, then dispose the request object.
+                // No point keeping it alive, just chewing resources.
+                if (e === this.cell.document) {
+                    this.request?.dispose(); // NOSONAR
+                }
+                if (this.started && !this._completed) {
+                    this.completedDueToCancellation().catch((ex) =>
+                        traceInfo('Failures when cancelling due to cell removal', ex)
+                    );
+                }
+            },
+            this,
+            disposables
+        );
     }
 
     public static fromCell(
-        editor: VSCNotebookEditor,
         cell: NotebookCell,
         errorHandler: IDataScienceErrorHandler,
         editorProvider: INotebookEditorProvider,
```

Comment:
This is only used to fire an event, with the new Execution API coming in, this dependency will go out completely.